code/game/objects/items/weapons/storage/pouches.dm

```diff
@@ -0,0 +1,212 @@
+	name = "medium generic pouch"
+	desc = "Can hold anything in it, but only about twice."
+	icon_state = "medium_generic"
+	item_state = "medium_generic"
+
+	max_storage_space = ITEM_SIZE_TINY * 4
+	max_w_class = ITEM_SIZE_SMALL
+
+/obj/item/weapon/storage/pouch/medical_supply
+	name = "medical supply pouch"
+	desc = "Can hold medical equipment. But only about three pieces of it."
+	icon_state = "medical_supply"
+	item_state = "medical_supply"
+
+	storage_slots = 3
+	max_w_class = ITEM_SIZE_SMALL
+
+	can_hold = list(
+		/obj/item/device/scanner/healthanalyzer,
+		/obj/item/weapon/dnainjector,
+		/obj/item/weapon/reagent_containers/dropper,
+		/obj/item/weapon/reagent_containers/glass/beaker,
+		/obj/item/weapon/reagent_containers/glass/bottle,
+		/obj/item/weapon/reagent_containers/pill,
+		/obj/item/weapon/reagent_containers/syringe,
+		/obj/item/weapon/storage/pill_bottle,
+		/obj/item/stack/medical,
+		/obj/item/clothing/mask/surgical,
+		/obj/item/clothing/head/surgery,
+		/obj/item/clothing/gloves/latex,
+		/obj/item/weapon/reagent_containers/hypospray,
+		/obj/item/clothing/glasses/hud/health,
+		)
+
+/obj/item/weapon/storage/pouch/engineering_tools
+	name = "engineering tools pouch"
+	desc = "Can hold engineering tools. But only about three pieces of them."
+	icon_state = "engineering_tool"
+	item_state = "engineering_tool"
+
+	storage_slots = 2
+	max_w_class = ITEM_SIZE_SMALL
+
+	can_hold = list(
+		/obj/item/weapon/tool,
+		/obj/item/device/lighting/toggleable/flashlight,
+		/obj/item/stack/cable_coil,
+		/obj/item/device/t_scanner,
+		/obj/item/device/scanner/analyzer,
+		/obj/item/taperoll/engineering,
+		/obj/item/device/robotanalyzer,
+		/obj/item/device/scanner/analyzer/plant_analyzer,
+		/obj/item/weapon/extinguisher/mini
+		)
+
+/obj/item/weapon/storage/pouch/engineering_supply
+	name = "engineering supply pouch"
+	desc = "Can hold engineering equipment. But only about two pieces of it."
+	icon_state = "engineering_supply"
+	item_state = "engineering_supply"
+
+	storage_slots = 2
+	w_class = ITEM_SIZE_NORMAL
+	max_w_class = ITEM_SIZE_NORMAL
+
+	can_hold = list(
+		/obj/item/weapon/cell,
+		/obj/item/weapon/circuitboard,
+		/obj/item/weapon/tool,
+		/obj/item/stack/material,
+		/obj/item/weapon/material,
+		/obj/item/device/lighting/toggleable/flashlight,
+		/obj/item/stack/cable_coil,
+		/obj/item/device/t_scanner,
+		/obj/item/device/scanner/analyzer,
+		/obj/item/taperoll/engineering,
+		/obj/item/device/robotanalyzer,
+		/obj/item/device/scanner/analyzer/plant_analyzer,
+		/obj/item/weapon/extinguisher/mini
+		)
+
+/obj/item/weapon/storage/pouch/ammo
+	name = "ammo pouch"
+	desc = "Can hold ammo magazines and bullets, not the boxes though."
+	icon_state = "ammo"
+	item_state = "ammo"
+
+	storage_slots = 3
+	w_class = ITEM_SIZE_NORMAL
+	max_w_class = ITEM_SIZE_NORMAL
+
+	can_hold = list(
+		/obj/item/ammo_magazine,
+		/obj/item/ammo_casing
+		)
+
+/obj/item/weapon/storage/pouch/flare
```

Comment:
Since it's very specific and niche purpose, make the flare pouch start pre-filled

In its Initialize proc, create five flares inside it.
Do it like...
```/obj/item/weapon/storage/pouch/flare/Initialize(var/roundstart)
if (roundstart)//make the flares```

Reply:
There was also a suggestion on allowing glow sticks and syringes be kept in these. Should I?

Reply:
yes this is actually an excellent idea
Syringes, autoinjectors, glowsticks, hypospray, pens/crayons and vials! this will give it a significant medical use

In that case scrap the prespawned flare idea, and rename it something more generic like Cylinder pouch